healthcheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+namespace Automattic\VIP;
+
+use Automattic\VIP\Utils\Context;
+/**
+ * Utility Healthcheck class that runs further than cache-healthcheck and allow to verify the app code is working.
+ */
+class Healthcheck {
+	/**
+	 * By default the site is considered healthy unless filter value returns falsy value
+	 *
+	 * @var boolean
+	 */
+	protected bool $healthy = true;
+	public function __construct() {
+		$this->check();
+	}
+	public function check() {
+		$this->healthy = (bool) apply_filters( 'vip_site_healthcheck', $this->healthy );
+	}
+
+	public function is_healthy() {
+		return $this->healthy;
+	}
+}
+
+/**
+ * It's safe to assume that by the time of `template_redirect` the codebase is loaded fully.
+ */
+if ( isset( $_SERVER['REQUEST_URI'] ) && '/vip-healthcheck' === $_SERVER['REQUEST_URI'] ) {
```

Comment:
Can the `'/vip-healthcheck'` string be defined as a constant of class? This way if the endpoint ever changes, we can just change the constant instead of touching the logic.

I think the code would look nicer if it was:

```php
if ( isset( $_SERVER['REQUEST_URI'] ) && VIP_HEALTHCHECK_ENDPOINT === $_SERVER['REQUEST_URI'] ) {
```

I admit this is a pedantic preference, I just think it looks better.

Reply:
Yeah, I personally like the inline hardcoded strings in cases like these (where they are used only once), which makes searching easier. IF there were two or more instances of using it, I'd agree with you.

Reply:
fair enough.